packages/amplify-cli-core/src/feature-flags/featureFlagFileProvider.ts

```diff
@@ -0,0 +1,79 @@
+import * as fs from 'fs-extra';
+import * as _ from 'lodash';
+import * as path from 'path';
+import { FeatureFlags, FeatureFlagsEntry } from '.';
+import { CLIEnvironmentProvider, JSONUtilities } from '..';
+import { FeatureFlagValueProvider } from './featureFlagValueProvider';
+import { AmplifyConfigFileName, AmplifyConfigEnvFileNameTemplate } from '../constants';
+
+export type FeatureFlagFileProviderOptions = {
+  projectPath?: string;
+};
+
+const defaultFeatureFlagFileProviderOptions: FeatureFlagFileProviderOptions = {};
+
+export class FeatureFlagFileProvider implements FeatureFlagValueProvider {
+  constructor(
+    private environmentProvider: CLIEnvironmentProvider,
```

Comment:
seems like this class could just take in the env name directly and get rid of all the ceremony with the CLIEnvironmentProvider and the CLIContextEnvironmentProvider

Reply:
This is a typical accessor pattern as this class cannot depend on when  will it be initialized, perhaps CLI will need features for uninitialized CLI project and such.